packages/react-native-renderer/src/NativeMethodsMixinUtils.js

```diff
@@ -11,26 +11,36 @@
  * In the future, we should cleanup callbacks by cancelling them instead of
  * using this.
  */
-export function mountSafeCallback(context: any, callback: ?Function): any {
+export function mountSafeCallback_NOT_REALLY_SAFE(
+  context: any,
+  callback: ?Function,
+): any {
   return function() {
     if (!callback) {
       return undefined;
     }
+    // This protects against createClass() components.
+    // We don't know if there is code depending on it.
+    // We intentionally don't use isMounted() because even accessing
+    // isMounted property on a React ES6 class will trigger a warning.
     if (typeof context.__isMounted === 'boolean') {
-      // TODO(gaearon): this is gross and should be removed.
-      // It is currently necessary because View uses createClass,
-      // and so any measure() calls on View (which are done by React
-      // DevTools) trigger the isMounted() deprecation warning.
       if (!context.__isMounted) {
         return undefined;
       }
-      // The else branch is important so that we don't
-      // trigger the deprecation warning by calling isMounted.
-    } else if (typeof context.isMounted === 'function') {
-      if (!context.isMounted()) {
```

Comment:
My hypothesis is we never actually hit this.

ES6 class components always return `undefined` from `isMounted` and log a warning.

RN host components don't have a method called `isMounted()`.

I guess a *user* could define their own `isMounted()` implementation but it wouldn't make sense to intentionally support this.